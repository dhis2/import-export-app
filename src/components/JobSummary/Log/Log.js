import i18n from '@dhis2/d2-i18n'
import {
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    TableRowHead,
    TableCellHead,
} from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { jsDateToString } from '../../../utils/helper.js'
import { FormField } from '../../index.js'

const Log = ({ events }) => {
    return (
        <FormField
            label={`${i18n.t('Log')}`}
            dataTest="job-summary-log"
            name="log"
        >
            <Table>
                <TableHead>
                    <TableRowHead>
                        <TableCellHead>{i18n.t('Time')}</TableCellHead>
                        <TableCellHead>{i18n.t('Message')}</TableCellHead>
                        <TableCellHead>{i18n.t('ID')}</TableCellHead>
                    </TableRowHead>
                </TableHead>
                <TableBody>
                    {(events || null) &&
                        events.map((e, i) => (
                            <TableRow key={`job-summary-log-${e.id}-${i}`}>
                                <TableCell>{jsDateToString(e.date)}</TableCell>
                                <TableCell>{e.text}</TableCell>
                                <TableCell>{e.id}</TableCell>
                            </TableRow>
                        ))}
                </TableBody>
            </Table>
        </FormField>
    )
}

const eventPropType = PropTypes.shape({
    date: PropTypes.instanceOf(Date).isRequired,
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
})

Log.propTypes = {
    events: PropTypes.arrayOf(eventPropType),
}

export { Log }
