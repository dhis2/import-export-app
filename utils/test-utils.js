import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
