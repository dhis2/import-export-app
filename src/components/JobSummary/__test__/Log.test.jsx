import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { Log } from '../Log/Log.jsx'

const props = {
    events: [
        {
            date: new Date('2020-01-01T00:00:00'),
            text: 'First day of year',
            id: 'Jan',
        },
        {
            date: new Date('2020-01-02T00:00:00'),
            text: 'Snd day of year',
            id: 'Jan2',
        },
        {
            date: new Date('2020-12-31T00:00:00'),
            text: 'Last day of year',
            id: 'Dec',
        },
    ],
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<Log {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
