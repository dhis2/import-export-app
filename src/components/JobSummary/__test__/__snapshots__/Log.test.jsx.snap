// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-log"
  >
    <span
      class="label"
    >
      Log
    </span>
    <table
      class="jsx-2430604489 "
      data-test="dhis2-uicore-table"
    >
      <thead
        data-test="dhis2-uicore-tablehead"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerowhead"
        >
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Time
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Message
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            ID
          </th>
        </tr>
      </thead>
      <tbody
        data-test="dhis2-uicore-tablebody"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            2020-01-01 00:00:00

          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            First day of year
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            Jan
          </td>
        </tr>
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            2020-01-02 00:00:00

          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            Snd day of year
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            Jan2
          </td>
        </tr>
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            2020-12-31 00:00:00

          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            Last day of year
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            Dec
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
